File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int size = 0;
        int n = s.size() - 1;

        for (int i = n; i >= 0; i--) {
            if (isalpha(s[i])) {
                ++size;
            } else if (size > 0 && !isalpha(s[i])) {
                i = -1;
            }
        }
        return size;
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        s = s.strip()
        n = len(s) - 1
        count = 0
        while n >= 0:
            if s[n] == ' ':
                return count
            else:
                count += 1
            n -= 1
        return count


